google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/GoogleComputeEngineServiceAdapter.java

```diff
@@ -173,6 +181,18 @@
       // Add lookup for InstanceToNodeMetadata
       diskToSourceImage.put(instance.get().disks().get(0).source(), template.getImage().getUri());
 
+      boolean windows = options.autoCreateWindowsPassword() != null
+    		  ? options.autoCreateWindowsPassword()
+    	      : (OsFamily.WINDOWS == template.getImage().getOperatingSystem().getFamily());
```

Comment:
Isn't here a need to always check the OS family? What happens if the `autoCreteWindowsPassword` is set and the image is, say, an Ubuntu?